frontend/src/metabase/search/components/InfoText.jsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { t, jt } from "ttag";
+
+import * as Urls from "metabase/lib/urls";
+import { capitalize } from "metabase/lib/formatting";
+
+import Icon from "metabase/components/Icon";
+import Link from "metabase/components/Link";
+
+import Schema from "metabase/entities/schemas";
+import Database from "metabase/entities/databases";
+import Table from "metabase/entities/tables";
+import { PLUGIN_COLLECTIONS } from "metabase/plugins";
+import { CollectionBadge } from "./CollectionBadge";
+
+function formatCollection(collection) {
+  return collection.id && <CollectionBadge collection={collection} />;
+}
+
+function getCollectionInfoText(collection) {
+  if (PLUGIN_COLLECTIONS.isRegularCollection(collection)) {
+    return t`Collection`;
+  }
+  const level = PLUGIN_COLLECTIONS.AUTHORITY_LEVEL[collection.authority_level];
+  return `${level.name} ${t`Collection`}`;
+}
+
+export function InfoText({ result }) {
+  const collection = result.getCollection();
+  switch (result.model) {
+    case "card":
+      return jt`Saved question in ${formatCollection(collection)}`;
+    case "collection":
+      return getCollectionInfoText(result.collection);
+    case "database":
+      return t`Database`;
+    case "table":
+      return (
+        <span>
```

Comment:
This entire block could be extracted as a component, making the switch statement faster to understand.

Reply:
I'll address it in the next PR to avoid conflicts